src/w3iProxy/chatProviders/internalChatProvider.ts

```diff
@@ -242,21 +242,30 @@ export default class InternalChatProvider implements W3iChatProvider {
       throw new Error(this.formatClientRelatedError('register'))
     }
 
-    console.log('internalChatProvider > register > account:', params.account)
-
-    return this.chatClient.register({
-      ...params,
-      onSign: async message => {
-        this.emitter.emit('chat_signature_requested', { message })
-
-        return new Promise(resolve => {
-          this.emitter.on('chat_signature_delivered', ({ signature }: { signature: string }) => {
-            console.log('Signature: ', signature)
-            resolve(signature)
+    try {
+      const registeredIdentityKey = await this.chatClient.register({
+        ...params,
+        onSign: async message => {
+          this.emitter.emit('chat_signature_requested', { message })
+
+          return new Promise(resolve => {
+            this.emitter.on('chat_signature_delivered', ({ signature }: { signature: string }) => {
+              resolve(signature)
+            })
           })
-        })
-      }
-    })
+        }
+      })
+
+      return registeredIdentityKey
+    } catch (e) {
+      /*
+       * Right now, chat client's register doesn't work 100% of the time
```

Comment:
Thanks for the explanation and the todo. Let's please create an issue in chatClient 🙏 

Going to be a lot more watchful of our error handling in w3i in reviews going forward, unhandled promise rejections have caused hard-to-understand issues a bunch of times now, primarily wasting your time in particular in the process.

Reply:
Likewise, if it was properly logged that w3i init was throwing this would've been caught ages ago. Thanks for the below suggestion

Reply:
https://github.com/WalletConnect/chat-client-js/issues/68